convict error when trying to clone and run from GitHub

### Steps to repro

``` sh
$ git clone git@github.com:mozilla/chronicle.git
$ cd chronicle
$ npm install
$ cp config/local.json.example config/local.json
$ npm start
```
### Actual results

```
> mozilla-chronicle@0.1.0 start /Users/pdehaan/dev/tmp/chronicle
> node server/index.js


/Users/pdehaan/dev/tmp/chronicle/node_modules/convict/lib/convict.js:393
        throw new Error(errBuf);
              ^
Error: server.session.duration: must be a positive integer: value was "7 days"
    at Object.rv.validate (/Users/pdehaan/dev/tmp/chronicle/node_modules/convict/lib/convict.js:393:15)
    at Object.<anonymous> (/Users/pdehaan/dev/tmp/chronicle/server/config.js:183:6)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/pdehaan/dev/tmp/chronicle/server/index.js:8:14)
    at Module._compile (module.js:456:26)

npm ERR! mozilla-chronicle@0.1.0 start: `node server/index.js`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the mozilla-chronicle@0.1.0 start script.
npm ERR! This is most likely a problem with the mozilla-chronicle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server/index.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls mozilla-chronicle
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 12.5.0
npm ERR! command "node" "/usr/local/bin/npm" "start"
npm ERR! cwd /Users/pdehaan/dev/tmp/chronicle
npm ERR! node -v v0.10.33
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/pdehaan/dev/tmp/chronicle/npm-debug.log
npm ERR! not ok code 0
```
### Expected results

No errors.

Need to dig in a bit more, but I think the problem w/ convict is this in the config/local.json.example file:

``` js
    "session": {
      "password": "Wh4t3ver.",
      "isSecure": false,
      "duration": "7 days"
    },
```

I think convict may want that duration in milliseconds here in the JSON file (but oddly accepts it as a valid default in server/config.js).

--- a/config/local.json.example
+++ b/config/local.json.example
@@ -31,7 +31,7 @@
     "session": {
       "password": "Wh4t3ver.",
       "isSecure": false,
-      "duration": "7 days"
+      "duration": 86400000
     },
     "oauth": {
       "clientId": "1f9bbddcb3e160ab",

--- a/server/config.js
+++ b/server/config.js
@@ -9,6 +9,8 @@ var path = require('path');
 
 var convict = require('convict');
 
+var SEVEN_DAYS_IN_MSEC = 1000 * 60 * 60 * 24 * 7; // 7 days
+
 var conf = convict({
   env: {
     doc: 'Application environment.',
@@ -20,49 +22,59 @@ var conf = convict({
     elasticsearch: {
       host: {
         default: '127.0.0.1',
-        env: 'ES_HOST'
+        env: 'ES_HOST',
+        format: 'url'
       },
       port: {
         default: 9200,
-        env: 'ES_PORT'
+        env: 'ES_PORT',
+        format: 'port'
       }
     },
     mysql: {
       host: {
         default: '127.0.0.1',
-        env: 'MYSQL_HOST'
+        env: 'MYSQL_HOST',
+        format: 'url'
       },
       port: {
         default: 3306,
-        env: 'MYSQL_PORT'
+        env: 'MYSQL_PORT',
+        format: 'port'
       },
       user: {
         default: 'root',
-        env: 'MYSQL_USERNAME'
+        env: 'MYSQL_USERNAME',
+        format: String
       },
       password: {
         default: '',
-        env: 'MYSQL_PASSWORD'
+        env: 'MYSQL_PASSWORD',
+        format: String
       },
       database: {
         doc: 'Name of the database.',
         default: 'chronicle',
-        env: 'MYSQL_DATABASE'
+        env: 'MYSQL_DATABASE',
+        format: String
       },
       connectionLimit: {
         doc: 'Max number of MySQL workers in the pool.',
         default: 25,
-        env: 'MYSQL_CONN_LIMIT'
+        env: 'MYSQL_CONN_LIMIT',
+        format: 'int'
       }
     },
     redis: {
       host: {
         default: '127.0.0.1',
-        env: 'REDIS_HOST'
+        env: 'REDIS_HOST',
+        format: 'url'
       },
       port: {
         default: 6379,
-        env: 'REDIS_PORT'
+        env: 'REDIS_PORT',
+        format: 'port'
       }
     }
   },
@@ -71,86 +83,103 @@ var conf = convict({
       doc: 'Mozlog configuration. Defaults currently target dev, not prod.',
       app: {
         doc: 'Top-level app name used in hierarchical logging, for example, "chronicle.db.mysql"',
-        default: 'chronicle'
+        default: 'chronicle',
+        format: String
       },
       level: {
         doc: 'Log messages with this or greater log level. Levels, in order, are: ' +
              'trace, verbose, debug, info, warn, error, critical.',
-        default: 'debug'
+        default: 'debug',
+        format: ['trace', 'verbose', 'debug', 'info', 'warn', 'error', 'critical']
       },
       fmt: {
         doc: 'Formatter. Use pretty for local dev, heka for production',
-        default: 'pretty'
+        default: 'pretty',
+        format: ['pretty', 'heka']
       },
       debug: {
         doc: 'Extra checks that logger invocations are correct. Useful when developing',
-        default: false
+        default: false,
+        format: Boolean
       }
     },
     host: {
       doc: 'Host for the main api server process.',
       default: '127.0.0.1',
+      format: 'url',
       env: 'HOST'
     },
     port: {
       doc: 'Port for the main api server process.',
       default: 8080,
+      format: 'port',
       env: 'PORT'
     },
     staticPath: {
       doc: 'Path to static files.',
       default: 'dist',
+      format: String,
       env: 'STATIC_PATH'
     },
     staticDirListing: {
       doc: 'Whether to allow directory listings for static dirs.',
       default: false,
+      format: Boolean,
       env: 'STATIC_DIR_LISTING'
     },
     oauth: {
       clientId: {
         doc: 'Firefox Accounts client_id.',
         default: '5901bd09376fadaa',
+        format: String,
         env: 'FXA_OAUTH_CLIENT_ID'
       },
       clientSecret: {
         doc: 'Firefox Accounts OAuth client_secret.',
         default: '4ab433e31ef3a7cf7c20590f047987922b5c9ceb1faff56f0f8164df053dd94c',
+        format: String,
         env: 'FXA_OAUTH_CLIENT_SECRET'
       },
       scope: {
         doc: 'Application scope. Currently not checked by the auth server.',
         default: ['chronicle', 'profile'],
+        format: Array,
         env: 'FXA_OAUTH_SCOPE'
       },
       version: {
         doc: 'OAuth version.',
         default: '2.0',
+        format: String,
         env: 'FXA_OAUTH_VERSION'
       },
       protocol: {
         doc: 'OAuth authorization protocol used.',
         default: 'oauth2',
+        format: String,
         env: 'FXA_OAUTH_AUTH_PROTOCOL'
       },
       redirectEndpoint: {
         doc: 'App endpoint to which OAuth flow redirects.',
         default: 'http://localhost:8080/auth/complete',
+        format: 'url',
         env: 'FXA_OAUTH_REDIRECT_ENDPOINT'
       },
       authEndpoint: {
         doc: 'Firefox Accounts OAuth authorization endpoint.',
         default: 'https://oauth-latest.dev.lcip.org/v1/authorization',
+        format: 'url',
         env: 'FXA_OAUTH_AUTH_ENDPOINT'
       },
       tokenEndpoint: {
         doc: 'Firefox Accounts OAuth token endpoint.',
         default: 'https://oauth-latest.dev.lcip.org/v1/token',
+        format: 'url',
         env: 'FXA_OAUTH_TOKEN_ENDPOINT'
       },
       profileEndpoint: {
         doc: 'Firefox Accounts profile server endpoint, used to fetch email, fxa_id.',
         default: 'https://latest.dev.lcip.org/profile/v1/profile',
+        format: 'url',
         env: 'FXA_PROFILE_ENDPOINT'
       }
     },
@@ -158,17 +187,19 @@ var conf = convict({
       password: {
         doc: 'Password used to encrypt cookies with hueniverse-iron.',
         default: 'S3kr1t',
+        format: String,
         env: 'SESSION_PASSWORD'
       },
       isSecure: {
         doc: 'Whether to allow cookie to be transmitted over insecure connections.',
         default: false,
+        format: Boolean,
         env: 'SESSION_ALLOW_INSECURE_COOKIES'
       },
       duration: {
         doc: 'Default session TTL for authenticated users.',
-        format: 'duration',
-        default: '7 days',
+        format: 'int',
+        default: SEVEN_DAYS_IN_MSEC,
         env: 'SESSION_DURATION'
       }
     }

--- a/server/index.js
+++ b/server/index.js
@@ -62,7 +62,6 @@ server.register([require('hapi-auth-cookie'), require('bell')], function (err) {
       log.warn('server failed to start: ' + err);
       throw err; // TODO: should we fail to start in some other way? AppError?
     }
-    log.info('chronicle server running on ' + config.get('server.host') +
-              ':' + config.get('server.port'));
+    log.info('chronicle server running on ' + server.info.uri);
   });
 });

